pkg/cluster/cluster.go

```diff
@@ -138,6 +138,16 @@ func (c *Cluster) prepareBackupAndRestore() error {
 		}
 	}
 
+	// We use backup RS to mark that we have done backup creation and even restore process before.
```

Comment:
We need to check backup unset case first:
```
	if backup == nil {
		if restore != nil {
			return errBackupUnsetRestoreSet
		} else {
			return nil
		}
	}
```

I feel like if we move them all outside, it will bloat the new() method.